src/wayland/single_pixel_buffer/mod.rs

```diff
@@ -0,0 +1,98 @@
+use wayland_protocols::wp::single_pixel_buffer::v1::server::wp_single_pixel_buffer_manager_v1::WpSinglePixelBufferManagerV1;
+use wayland_server::{
+    backend::GlobalId, protocol::wl_buffer::WlBuffer, Dispatch, DisplayHandle, GlobalDispatch, Resource,
+};
+
+mod handlers;
+
+/// Delegate state of WpSinglePixelBuffer protocol
+#[derive(Debug)]
+pub struct SinglePixelBufferState {
+    global: GlobalId,
+}
+
+impl SinglePixelBufferState {
+    /// Create a new [`WpSinglePixelBufferManagerV1`] global
+    //
+    /// The id provided by [`SinglePixelBufferState::global`] may be used to
+    /// remove or disable this global in the future.
+    pub fn new<D>(display: &DisplayHandle) -> Self
+    where
+        D: GlobalDispatch<WpSinglePixelBufferManagerV1, ()>,
+        D: Dispatch<WpSinglePixelBufferManagerV1, ()>,
+        D: 'static,
+    {
+        let global = display.create_global::<D, WpSinglePixelBufferManagerV1, _>(1, ());
+
+        Self { global }
+    }
+
+    /// Returns the id of the [`WpSinglePixelBufferManagerV1`] global.
+    pub fn global(&self) -> GlobalId {
+        self.global.clone()
+    }
+}
+
+/// User data of `WlBuffer` backed by single pixel
+#[derive(Debug)]
+pub struct SinglePixelBufferUserData {
+    /// Value of the buffer's red channel
+    pub r: u32,
+    /// Value of the buffer's green channel
+    pub g: u32,
+    /// Value of the buffer's blue channel
+    pub b: u32,
+    /// Value of the buffer's alpha channel
+    pub a: u32,
+}
+
+impl SinglePixelBufferUserData {
+    /// Check if pixel has alpha
+    pub fn has_alpha(&self) -> bool {
+        self.a != u32::MAX
+    }
+
+    /// RGAB8888 color buffer
+    pub fn rgba8888(&self) -> [u8; 4] {
+        let divider = u32::MAX / 255;
```

Comment:
divisor is technically the correct term here.

Reply:
fixed